packages/squiggle-lang/src/rescript/ReducerProject/ReducerProject_ProjectItem_T.res

```diff
@@ -0,0 +1,39 @@
+module Parse = Reducer_Peggy_Parse
+module ExpressionT = Reducer_Expression_T
+module InternalExpressionValue = ReducerInterface_InternalExpressionValue
+module ExternalExpressionValue = ReducerInterface_ExternalExpressionValue
+open Reducer_ErrorValue
+
+type sourceArgumentType = string
+type sourceType = string
+type rawParseArgumentType = result<Parse.node, errorValue>
+type rawParseType = option<rawParseArgumentType>
+type expressionArgumentType = result<ExpressionT.t, errorValue>
+type expressionType = option<expressionArgumentType>
+type continuation = InternalExpressionValue.nameSpace
+type continuationArgumentType = InternalExpressionValue.nameSpace
+type continuationType = option<continuationArgumentType>
+type continuationResultType = option<result<continuationArgumentType, errorValue>>
+type externalBindingsArgumentType = ExternalExpressionValue.record
+type externalBindingsType = option<externalBindingsArgumentType>
+type resultArgumentType = result<InternalExpressionValue.t, errorValue>
+type resultType = option<resultArgumentType>
+type externalResultArgumentType = result<ExternalExpressionValue.t, errorValue>
+type externalResultType = option<externalResultArgumentType>
+type continuesArgumentType = array<string>
+type continuesType = array<string>
+type includesArgumentType = string
+type includesType = result<array<string>, errorValue>
+
+type projectItem =
+  | ProjectItem({
```

Comment:
I take it you have some reason for this being a variant?

Reply:
> I think this could really use a Readme.md, just to outline how the `ReducerProject` works and how to interact with it. The interface is really big now, and there are some terms I don't really understand.

Explaining methods one by one is not very efficient. 
The unit test serves as the example and the tutorial.
If there needs to more clarity about any method then that means I should make yet another test case to illustrate the use.

Reply:
> If there needs to more clarity about any method then that means I should make yet another test case to illustrate the use.

I disagree here. For example, when open source packages don't have READMEs, but just say, "Look at the code, like the tests", typically that's a pretty bad sign to me. Having tests is great, but it's very difficult for them to explain everything you need, especially for a complicated setup like this. 

I'd think of the interpreter a bit like a separate npm package in understanding how complicated it is. 

Part of the issue might be that the tests seem pretty simple to me. I'm not really sure how to connect the information in the tests with understanding how use of the library will change, which seems like it could be a really big deal. (So I care about understanding it at lot)